wrappers/csharp/src/Frame.cs

```diff
@@ -0,0 +1,221 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Intel.RealSense
+{
+    public class Frame : IDisposable
+    {
+        internal HandleRef m_instance;
+
+        public Frame(IntPtr ptr)
+        {
+            //if (ptr == IntPtr.Zero)
+            //    throw new ArgumentNullException("ptr");
+            m_instance = new HandleRef(this, ptr);
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    // TODO: dispose managed state (managed objects).
+                }
+
+                // TODO: free unmanaged resources (unmanaged objects) and override a finalizer below.
+                // TODO: set large fields to null.
+                Release();
+                disposedValue = true;
+            }
+        }
+
+        // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
+        ~Frame()
+        {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+            Dispose(false);
+        }
+
+        // This code added to correctly implement the disposable pattern.
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+            Dispose(true);
+            // TODO: uncomment the following line if the finalizer is overridden above.
+            GC.SuppressFinalize(this);
+        }
+        #endregion
+
+        public void Release()
+        {
+            if(m_instance.Handle != IntPtr.Zero)
+                NativeMethods.rs2_release_frame(m_instance.Handle);
+            m_instance = new HandleRef(this, IntPtr.Zero);
+        }
+
+        public Frame Clone()
+        {
+            object error;
+            NativeMethods.rs2_frame_add_ref(m_instance.Handle, out error);
+            return new Frame(m_instance.Handle);
+        }
+
+        public int Width
```

Comment:
Shouldn't this be part of VideoFrame? (btw, It's working now because you only use images)